grype/db/curator.go

```diff
@@ -267,6 +278,24 @@ func (c *Curator) download(listing *ListingEntry, downloadProgress *progress.Man
 	return tempDir, nil
 }
 
+func (c *Curator) validateStaleness(m *Metadata) error {
+	if m == nil {
+		return nil
+	}
+
+	age := time.Since(m.Built).Round(time.Hour)
```

Comment:
What is the motivation for rounding here?

Reply:
I added a code comment explaining it; 

tl;dr time comparison has fractional seconds of precision which makes the error message noisy. Rounding it to hour provides enough context for action.

Reply:
Rounding still happens but through a time formatting library capable of printing longer durations in a more friendly way, like `22 days` instead of `528h`